aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,217 @@
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/project.json
+  :language: javascript
+  :lines: 4-26
+  :emphasize-lines: 8
+  :dedent: 2
+
+Next, add the ``XmlSerializerOutputFormatter`` to MVC's list of OutputFormatters in *Startup.cs*:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Startup.cs
+  :language: c#
+  :lines: 30-39
+  :emphasize-lines: 6
+  :dedent: 8
+
+Alternately, you can just call the ``AddXmlSerializerFormatters`` extension method:
+
+.. code-block:: c#
+  :emphasize-lines: 2
+
+  services.AddMvc()
+    .AddXmlSerializerFormatters();
+
+Once you've added support for XML formatting, your API methods should return the appropriate format based on the request's ``Accept`` header, as this Fiddler example demonstrates:
+
+.. image:: formatting/_static/xml-response.png
+
+You can see at the top of the image that the GET request was made with an ``Accept: application/xml`` header set. The response at the bottom of the image has a ``Content-Type: application/xml`` header, and you can see at the bottom the ``Author`` object has been serialized to XML.
+
+Making the same request, but with ``application/json`` specified in the ``Accept`` header, results in a JSON formatted response:
+
+.. image:: formatting/_static/json-response-fiddler.png
+
+In this screenshot, you can see the request sets a header of ``Accept: application/json`` and the response specifies the same as its ``Content-Type``. The ``Author`` object is shown in the body of the response, in JSON format.
+
+Forcing a Particular Format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you would like to require the use of a particular response format, without resorting to using a return type like ``JsonResult``, you can apply the ``[Produces]`` filter. Like most :doc:`/mvc/controllers/filters`, this can be applied at the action, controller, or global scope. When you specify it, you provide the content type that should be produced as a string:
+
+.. code-block:: c#
+
+  [Produces("application/json")]
+  public class AuthorsController
+
+The above filter would force all actions within the AuthorsController to return JSON-formatted responses, even if other formatters were configured for the application and the client provided an ``Accept`` header requesting a different, available format.
+
+Special Case Formatters
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Some special cases are implemented using built-in formatters, as already described above. By default, ``string`` return types will be formatted as `text/plain`. This behavior can be removed by removing the ``TextPlainFormatter``. Likewise, actions that have a model object return type will return a 204 No Content resposne when returning ``null``. This behavior can be removed by removing the ``HttpNoContentOutputFormatter``. Removing formatters is done in *Startup* when configuring MVC.
```

Comment:
after this, add 
the following code removes the TextPlainFormatter and HttpNoContentOutputFormatter